manifests/install.pp

```diff
@@ -30,6 +32,16 @@
       ensure => installed,
       before => File[$cert_dir, $conf_dir],
     }
+
+    if $manage_selinux {
+      package { $selinux_name:
+        ensure => installed,
+        after  => [Package[$package_name], File[$cert_dir, $conf_dir]],
+      } ~>
+      exec { '/usr/sbin/restorecon -R -F /var/lib/icinga2':
+        refreshonly => true
+      }
+    }
```

Comment:
Two things I have in mind here:

- Limit this to osfamily - redhat
- Why the manual restorecon exec?

Reply:
Limit to RedHat family isn't needed. SELinux is included since kernel 2.6 and not limited to RedHat systems. But of course the icinga project only distributes packages for RedHat family. Maybe this will change in the future.

We switched to module data in puppet-icinga2 v2.0 so you're able to use your own configured package. In this case it's important that the default fit to the operating system or is set to false.

Reply:
Since there is no package `icinga2-selinux` on distributions other than RHEL, I'd opt for the osfamily as otherwise the manifest will fail on Debian and alike. Even if you manage all that with a boolean up front.

Reply:
Default to false to all platforms (include Debian), the puppet run will be finished without any errors. If you set selinux to true you have to know what you're doing.

Starts with module data it isn't  accepted to set such limits anymore. Because now you can extend the module with your own packages or packages for not supported platform.